### PR TITLE
Add .flake8

### PR DESCRIPTION
**Pull request by zulcss**
Add flake8 configuration file so that we can start
to correct pep8 errors.

Signed-off-by: Charles Short <chucks@redhat.com>

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+exclude=.venv,.git,.tox,dist,doc,*lib/python*,*egg,build,*web-server/*,*agent/stockpile/*
+# For a full list see https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes
+# E1 - Indentation
+# E2 - Whitespace
+# E3 - Blank Line
+# E4 - Import
+# E5 - Line length
+# E7 - Statement
+# E9 - Runtime
+# W1 - Indenation warning
```
Comment (ndokos) [inline .flake8]: Indentation

Comment (zulcss) [reply .flake8]: Fixed

**Comment by zulcss** [comment]
> Good start!
> 
> Can we also add the pre-commit hook with this PR?
> 
> And consider changing the ignore to catch a few more problems? Something like:
> 
> ```
> ignore = E1, E2, E3, E4, E5, E6, W1, W2, W3, W4, W5, W6
> ```

I added F

> Good start!
> 
> Can we also add the pre-commit hook with this PR?
> 
> And consider changing the ignore to catch a few more problems? Something like:
> 
> ```
> ignore = E1, E2, E3, E4, E5, E6, W1, W2, W3, W4, W5, W6
> ```

I added F4, F5, F6 in a later commit.

**Comment by zulcss** [comment]
Added flake8 to travis and filtered out the 'H' errors as well.

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+This project uses the flake8 method of code style enforcement, linting and checking,
+ [flake8](http://flake8.pycqa.org/en/latest)
+
+All python code contributed to pbench must match the style requirements. These
+requirements are enforced by [pre-commit] (https://pre-commit.com).
+
+## Use pre-commit to set automatic commit requirements
+
+This project makes use of [pre-commit](https://pre-commit.com/) to do automatic
+lint and style checing on every commit containing Python files.
```
Comment (ndokos) [inline doc/CONTRIBUTING.md]: checking

Comment (zulcss) [reply doc/CONTRIBUTING.md]: Done

**Pull request merged**